osversion/platform_compat_windows.go

```diff
@@ -0,0 +1,71 @@
+package osversion
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// List of stable ABI compliant
+// ltsc releases
+// Note: List must be sorted
+var compatLTSCReleases = []uint{
+	V21H2Server,
+}
+
+// Checks if given host and guest OS versions are compatible.
+// It includes support for stable ABI compliant OS versions.
+// Every release after WS 2022 will support the previous
+// ltsc container image. Stable ABI is in preview mode
+// for windows 11 client.
+// Refer: https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2022%2Cwindows-10#windows-server-host-os-compatibility
+func AreHostAndGuestCompatible(hostOSVersion, guestOSVersion string) bool {
```

Comment:
Should we instead make this:

```
type CompatibleOptions struct {
    HypervisorIsolated bool
}

func WithHypervisorIsolatedOpt() {
   // ...
}

osversion.IsContainerImageCompatible(imageVersion, opts CompatibleOption...) bool {
    // if opts.HypervisorIsoalted verify rules for hypervisor isolation from 2016 to current
    // else verify rules for process isolation from 2016 to current. On 2016 and 2019 and 2022 must match minor dirctly. On Win11 different rules.
}
```

It seems like it would be very valuable to have these funcs correctly done in a common location outside of docker and containerd for both

Reply:
This function and compatibility checks are going to be the same for hyperv and process isolated. I am not sure I understand why we would need special options for hyperV case.
To get hyperV and multiarch images to work end to end we are working on adding runtime option to image pull (working on the prototype and KEP currently that should be out soon). Once image pulls have a runtime option, one can specify what version they intend to run a container image on. That is, for hyperV cases, this OS version got from runtime class will specify version of UVM and this version is what will be be passed the AreHostAndGuestCompatible() function from containerd. For process isolated, we would always pass the actual OS version of the host.

Reply:
cc @kevpar

Reply:
But they arent the same checks right? For xenon's, I can always run my image in the matching host because I can run that images utility container. For process isolated I have this rule chain to follow. Unless the intent of this method is to match for a complete pod where the method would then have to be the hosting os version of the xenon utility vm and then then image version would be the workload/sidecars? What is the intent here?

Reply:
Said differently, there are more checks for Xenon I think. There is:

1. Can I run a given UtilityVM version on a given Host version. There are ABI compatibility for the Bridge and thus needs to be verified for message compat. 
2. There is Can this UtilityVM Version run this Image (as a process isolated).

2, is the same as an Argon on a Windows host, but where are we verifying 1? Example, where is it supported to run a 2016 Xenon with 2016 container, on a Win 11 host?

Reply:
Yes, I agree this function should just validate the process-isolated host/guest compatibility. I don't think we currently have any mechanism built to validate host/guest UVM/bridge compatibility. That's something we may want to revisit as we work on bringing cross-version hv-isolated containers to CRI.